### PR TITLE
Replace all FontAwesome icons with svg icons #5535

### PR DESCRIPTION
**Pull request by elvisdorkenoo**
# Description
Replace all FontAwesome icons with svg icons [#5535](https://github.com/medic/cht-core/issues/5535)

[description]
Following the steps of[ these suggestions](https://github.com/medic/cht-core/issues/5535#issuecomment-1143840478)
medic/cht-core#5535

FA 4 -> FA 6 steps :
 - [ ] 1. Install FA 6 and make the webapp project run (testable)
 - [ ] 2. load and test FA 6 css icons  (testable)
 - [ ] 3. load and test FA 6 svg icons  (testable)
 - [ ] 4. Upgrade admin app FA 4 to FA 6 (not yet)
 - [ ] 5. Change the icons to the news ones (testable)
 - [ ] 6. Re-style/ customize svg icons to match our design (testable)
 - [ ] 7. Change all png custom icons to svg versions  (testable except logo icon.png)
 - [ ] 8. Upload our own custom icons to FA ? use FA Pro ? (not yet)
 - [ ] 9. Test and adjust(eventually) icons for mobile version(not yet) 
 - [ ] 10. Update e2e tests  (testable)
 - [ ] 11. Backwards compatibility, make sure nothing is broken or fix if something is broken, check svg icons loading and offline metrics  (testable)

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: webapp/src/css/inbox.less [hunk 1]
```diff
@@ -248,8 +248,10 @@ body {
           &, &:before {
             display: block;
             margin: auto;
-            height: 24px;
-            max-width: 24px;
+            height: 30px;
+            width: 30px;
+            max-width: 30px;
+            font-size: 30px;
```
Comment (latin-panda) [inline webapp/src/css/inbox.less:254]: is max-width and font-size still needed?

File: webapp/src/css/icon.less [hunk 2]
```diff
@@ -81,3 +93,161 @@
     transform: rotate(360deg);
   }
 }
+
+.fa-comments:before{
```
Comment (latin-panda) [inline webapp/src/css/icon.less:97]: You can [use parametric mixins](https://lesscss.org/features/#mixins-feature-mixins-parametric-feature) that helps you reusing code, the mixin can accept the icon url or colors as parameters. We have done this before in [variables.less](https://github.com/medic/cht-core/blob/master/webapp/src/css/variables.less)

**Review by latin-panda — CHANGES_REQUESTED** [review]
Thanks Elvis! 
Once we know if we're going with this option o FA 6, then I'll do a more detail review. 
But for know I added a comment about Less' parametric mixin that can fit here :)

**Review by latin-panda — CHANGES_REQUESTED** [review]
Nice! I added some questions in line.
Additionally, 
- I noticed that `admin` app is still using the old font-awesome.
- From your checkbox list in this PR description and ticket, what items are completed?
- I think we need to revert the change in `package-lock.json`, in `grunt-contrib-watch` section, it's giving me some issues when installing the project with `npm ci`
- Please remove the shadow in the icons 
Screen Shot 2022-06-20 at 1 28 18 pm

File: Gruntfile.js [hunk 3]
```diff
@@ -533,8 +533,8 @@ module.exports = function(grunt) {
             'patch webapp/node_modules/bootstrap-daterangepicker/daterangepicker.js < webapp/patches/bootstrap-daterangepicker.patch',
 
             // patch font-awesome to remove version attributes
-            // https://github.com/FortAwesome/Font-Awesome/issues/3286
-            'patch webapp/node_modules/font-awesome/less/path.less < webapp/patches/font-awesome-remove-version-attribute.patch',
+            // https://github.com/FortAwesome/Font-Awesome/issues/3286 (apparently fixed)
+            // 'patch webapp/node_modules/font-awesome/less/path.less < webapp/patches/font-awesome-remove-version-attribute.patch',
```
[the remark this answers is not in the document]
Comment (elvisdorkenoo) [reply Gruntfile.js:537]: confirmed and removed

File: package.json [hunk 4]
```diff
@@ -158,5 +158,7 @@
       "maxSize": "70 kB"
     }
   ],
-  "dependencies": {}
+  "dependencies": {
+    "@fortawesome/fontawesome-free": "^6.1.1"
```
[the remark this answers is not in the document]
Comment (elvisdorkenoo) [reply package.json:162]: I remove it from the main project and and keep them in both admin and webapp projects as it was before

yup, `"@fortawesome/fontawesome-free": "^6.1.1"` is the new way to distribute it, under `fortawesome` there are many fontawesome sub-dependencies as `fontawesome-free, fontawesome-pro`, etc...

File: package-lock.json [hunk 5]
```diff
@@ -15302,8 +15282,8 @@
       }
     },
     "grunt-contrib-watch": {
-      "version": "github:scdf/grunt-contrib-watch#896dba3fed7874ed85074f787a5baf60db80d12f",
-      "from": "github:scdf/grunt-contrib-watch#v1.0.0-fix_dirs_firing",
+      "version": "git+ssh://git@github.com/scdf/grunt-contrib-watch.git#896dba3fed7874ed85074f787a5baf60db80d12f",
```
[the remark this answers is not in the document]
Comment (elvisdorkenoo) [reply package-lock.json:15285]: I reverted the whole `package-lock.json` file

File: webapp/angular.json [hunk 6]
```diff
@@ -38,7 +38,9 @@
             },
             "scripts": [
               "node_modules/jquery/dist/jquery.min.js",
-              "node_modules/select2/dist/js/select2.min.js"
+              "node_modules/select2/dist/js/select2.min.js",
+              "node_modules/@fortawesome/fontawesome-free/js/all.js",
+              "node_modules/@fortawesome/fontawesome-free/js/v4-shims.js"
```
[the remark this answers is not in the document]
Comment (elvisdorkenoo) [reply webapp/angular.json:43]: _When using SVG + JS, you'll want to include the following v4-shims.js file In addition to whatever v6 JS files you're including. This file will map v4 icon names to icons in the latest version of Font Awesome._
more infos [here](https://fontawesome.com/docs/web/setup/upgrade/upgrade-from-v4)

File: webapp/package.json [hunk 7]
```diff
@@ -25,6 +25,7 @@
     "@angular/platform-browser": "^12.2.1",
     "@angular/platform-browser-dynamic": "^12.2.1",
     "@angular/router": "^12.2.1",
+    "@fortawesome/fontawesome-free": "^6.1.1",
```
[the remark this answers is not in the document]
Comment (elvisdorkenoo) [reply webapp/package.json:28]: I remove it from the main project and and keep them in both admin and webapp projects as it was before.

_Have you check if font-awesome 6 works in admin app?_
working on it, facing a bug with FA6 installed, Admin app just won’t load the icons and show small squares instead

File: webapp/src/css/inbox.less [hunk 8]
```diff
@@ -176,6 +175,9 @@ body {
 
       .mm-icon {
         color: @top-header-color;
```
[the remark this answers is not in the document]
Comment (elvisdorkenoo) [reply webapp/src/css/inbox.less:177]: removed

File: webapp/src/css/inbox.less [hunk 9]
```diff
@@ -197,6 +199,13 @@ body {
       margin-top: 0;
       white-space: normal;
       font-size: 1rem;
+      svg {
+        width: 20px;
+        height: 20px;
+        path {
+        fill : @top-header-color;
```
[the remark this answers is not in the document]
Comment (elvisdorkenoo) [reply webapp/src/css/inbox.less:206]: There's none for `color: #262626;`, the closest is `@gray-ultra-dark: #333333;`
created `@gray-extreme-dark: #262626;`

File: webapp/src/ts/components/actionbar/actionbar.component.html [hunk 10]
```diff
@@ -129,7 +129,7 @@
         <div *ngIf="selectMode">
           <div class="actions dropup" *ngIf="showActionBar && selectedReportsDocs?.length && (!loadingContent) && snapshotData?.name === 'reports.detail'">
             <a class="mm-icon mm-icon-inverse mm-icon-caption delete-all" mmAuth="can_bulk_delete_reports" (click)="bulkDelete(selectedReportsDocs)">
-              <span class="fa fa-trash-o"></span>
+              <span class="fa fa-trash-can"></span>
```
[the remark this answers is not in the document]
Comment (elvisdorkenoo) [reply webapp/src/ts/components/actionbar/actionbar.component.html:132]: fa-check-square-o are fa-square-o are no more fa6 icons name but thanks to v4-shims.js they will work just fine ; all the rest are ok. Same for 'fa' that should normally be 'fa-solid' or 'fas' if we have to strictly follow fa6 naming

[the remark this answers is not in the document]
Comment (latin-panda) [reply webapp/src/ts/components/actionbar/actionbar.component.html:132]: Oh I understand so `v4-shims.js` is their backward compatibility tool. Nice

[the remark this answers is not in the document]
Comment (elvisdorkenoo) [reply webapp/src/ts/components/actionbar/actionbar.component.html:132]: exactly !!

File: webapp/src/ts/components/header/header.component.html [hunk 11]
```diff
@@ -123,7 +123,7 @@
     <div class="tabs small-font">
       <a *ngFor="let tab of permittedTabs" routerLink="{{tab.route}}" id="{{tab.name}}-tab" class="{{tab.name}}-tab" [class.selected]="currentTab === tab.name">
         <div class="mm-icon" [class.mm-icon-inverse]="currentTab !== tab.name">
-          <span *ngIf="!tab.resourceIcon" class="fa {{tab.icon}}"><span class="fa {{tab.defaultIcon}}"></span></span>
+          <span *ngIf="!tab.resourceIcon" class="fa {{tab.icon === undefined ? tab.defaultIcon : tab.icon}}"></span>
```
[the remark this answers is not in the document]
Comment (elvisdorkenoo) [reply webapp/src/ts/components/header/header.component.html]: Yes it's much nicer, thanks for the suggestion !
For testing [header-tabs.service.spec.ts](https://github.com/medic/cht-core/webapp/tests/karma/ts/services/header-tabs.service.spec.ts) does it very well

File: webapp/src/css/icon.less (hunk 2 above)
Comment (elvisdorkenoo) [reply webapp/src/css/icon.less]: yeah, good idea, it's done.

**Comment by garethbowen** [comment]
@elvisdorkenoo I think we decided not to go ahead with this. Should we close this PR off?

**Pull request closed without merging**